/*
 * Licensed to Ververica GmbH under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
syntax = "proto3";

package org.apache.flink.statefun.examples.kafka;

import "google/protobuf/timestamp.proto";

option java_package = "com.ververica.statefun.workshop.generated";
option java_multiple_files = true;

message ConfirmFraud {
    string account = 1;
}

message Transaction {
    string account = 1;
    google.protobuf.Timestamp timestamp = 2;
    string merchant = 3;
    int32 amount = 4;
}

message FeatureVector {
    int32 fraudCount    = 1;
    int32 merchantScore = 2;
}

message FraudScore {
    int32 score = 1;
}
